adding physical_adapter check

vaishnavibhat opened the pull request:
adding physical_adapter param to check of the tested
interface is physical or virtual adapter

Signed-off-by: Vaishnavi Bhat <vaishnavi@linux.vnet.ibm.com>

vaishnavibhat commented:
[bonding.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/5331506/bonding.log)

avocado run bonding.py -m bonding_single.yaml 
JOB ID     : 6dcb35b6c856db3d4e3c196cb26c386380be01a3
JOB LOG    : /root/bond_scen/upstream/tests/results/job-2020-10-05T22.54-6dcb35b/job.log
 (1/3) bonding.py:Bonding.test_setup;run-d24e: PASS (20.00 s)
 (2/3) bonding.py:Bonding.test_run;run-d24e: PASS (261.85 s)
 (3/3) bonding.py:Bonding.test_cleanup;run-d24e: PASS (10.93 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/bond_scen/upstream/tests/results/job-2020-10-05T22.54-6dcb35b/results.html
JOB TIME   : 293.60 s

cforno12 commented on io/net/bonding.py:
```diff
@@ -286,9 +288,12 @@ def bond_fail(self, arg1):
-                    '8000', '9000']
-        for mtu in bond_mtu:
+        if self.physical_interface:
+            self.bond_mtu = ['2000', '3000', '4000', '5000', '6000', '7000',
```
I don't think we need to create a class variable `self.bond_mtu` as we are only using that array in this context (context being the bond_fail function). 

Other than that, LGTM!

cforno12 replied:
I think you're going to want to have two bond_mtu list variables. One for the host interface and one for the peer interface as you can have a virtual interface sending traffic to a physical and vice versa.

narasimhan-v commented on io/net/bonding.py:
```diff
@@ -112,6 +112,8 @@ def setUp(self):
         self.peer_wait_time = self.params.get("peer_wait_time", default=5)
         self.sleep_time = int(self.params.get("sleep_time", default=5))
         self.mtu = self.params.get("mtu", default=1500)
+        self.physical_interface = self.params.get("physical_interface",
+                                                  default=True)
```
@vaishnavibhat Sorry for looking into this very late.

Can you use a sysfs or ethtool command logic to decide whether an interface is physical or virtual please ?

abdhaleegit reviewed (changes requested):
@vaishnavibhat Please use sysfs param to decide if interface is physical or virtual

virtual
[root]# cat eth1/phys_port_name
cat: eth1/phys_port_name: Operation not supported

physical
[root]# cat eth2/phys_port_name
p0

I did not see a unique param to differentiate in ethtool

vaishnavibhat commented:
I can differentiate between a veth/vnic to a physical adapter . But the problem arises when I have a sriov adapter. But neverthless, for sriov adapter I intend to test all mtu. So I can make changes to read based on sysfs. 
Can I proceed with the approach of reading the /sys/class/net/ output ? virtual adapters have vio/ virtual tagged to it rather than the pci ids. Note, I could not find any sysfs way for sriov,, even though its virtual, it has pci id associated with it. 

lrwxrwxrwx. 1 root root 0 Nov 18 04:38 eth0 -> ../../devices/pci0013:01/0013:01:00.0/net/eth0
lrwxrwxrwx. 1 root root 0 Nov 18 04:38 eth1 -> ../../devices/pci0013:01/0013:01:00.1/net/eth1
lrwxrwxrwx. 1 root root 0 Nov 18 04:38 lo -> ../../devices/virtual/net/lo
lrwxrwxrwx. 1 root root 0 Nov 18 04:38 net0 -> ../../devices/vio/30000002/net/net0

cforno12 reviewed (changes requested):
You can also do `lsdevinfo -q name=<dev_name>` and get the value of driver, ibmveth or ibmvnic. i.e. 

```
# lsdevinfo -q name=eth0
device:
	name="eth0"
	uniquetype="adapter/vdevice/IBM,l-lan"
	class="adapter"
	subclass="vdevice"
	type="IBM,l-lan"
	prefix="eth"
	driver="ibmveth"
	status="1"

path:
	parent="vio"
	physloc="U9040.MR9.132987X-V2-C2-T1"
	connection="30000002"
```

narasimhan-v commented:
@vaishnavibhat Any updates on this ?

vaishnavibhat commented:
Updated the patch with lsdevinfo -q name=<dev_name> check and removed the physical_adapter parm from yaml files. 

[bonding_mtu.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/6110763/bonding_mtu.log)
]# avocado run bonding.py -m bonding.py.data/bonding_single.yaml 
JOB ID     : 25e1f493282d815357b5f09efcc6119b2128978d
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2021-03-09T13.39-25e1f49/job.log
 (1/3) bonding.py:Bonding.test_setup;run-0b73: PASS (150.04 s)
 (2/3) bonding.py:Bonding.test_run;run-0b73: PASS (158.12 s)
 (3/3) bonding.py:Bonding.test_cleanup;run-0b73: PASS (12.02 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2021-03-09T13.39-25e1f49/results.html
JOB TIME   : 321.26 s

abdhaleegit commented on io/net/bonding.py:
```diff
@@ -285,9 +296,13 @@ def bond_fail(self, arg1):
+            bond_mtu = ['9000']
+        else:
+            bond_mtu = ['2000', '3000', '4000', '5000', '6000', '7000',
+                        '8000', '9000']
```
@vaishnavibhat we could avoid else by

```
        bond_mtu = ['2000', '3000', '4000', '5000', '6000', '7000',
                        '8000', '9000']
        if self.is_vnic():
            bond_mtu = bond_mtu[-1]
```

abdhaleegit replied:
@vaishnavibhat Why not move is_vnic to utils ? this will be used in other tests too ?

abdhaleegit reviewed (approved):
LGTM
